### DISCUSSION TITLE
OpenAPIHono: defaultHook is not called on parent OpenAPIHono instance

### DISCUSSION THREAD

**Discussion by cldsnchz** [honojs/hono#1798] [Q&A] [answered]
Hi, when I define a OpenAPIHono with a defaultHook and a child route, like this:
```
const app = new OpenAPIHono({
  defaultHook: (result, c) => {
    if (!result.success) {
      return c.json({ message: result.error.message }, 422);
    }
  },
});

const users = new OpenAPIHono();

app.route("users", users);
```
The defaultHook is not called when there is an error in a path defined in `users`, is that the expected behavior? should I define the defaultHook on each OpenAPIHono instance?

**Comment by yusukebe** [comment, marked as answer]
> The defaultHook is not called when there is an error in a path defined in users, is that the expected behavior? should I define the defaultHook on each OpenAPIHono instance?

Yes, that is the expected behavior. You should define the `defaultHook` for each instance.
